### PR TITLE
Add a dataset bridging country names and ISO codes

### PR DESCRIPTION
**Pull request by jdhoffa**
Adding a dataset to bridge country names with their respective ISO codes. 
Useful for the analysis package

### CODE REVIEW

File: R/iso_codes.R
Comment (maurolepore) [inline R/iso_codes.R:5]: This `@family` will group this dataset with other [demo datasets](https://2degreesinvesting.github.io/r2dii.dataraw/reference/index.html#section-demo-datasets) but it looks like this dataset doesn't belong there -- as suggested by the lack of the suffix `_demo` in the dataset name.

Instead I think it is more closely related to the [`#' @family datasets for bridging sector classification codes`](https://2degreesinvesting.github.io/r2dii.dataraw/reference/index.html#section-datasets-to-look-up-sector-classification-codes). But that family name is too specific to include `iso_codes` so I suggest you change the `@family` to `#' @family datasets to look up codes` -- which would include datasets too look up iso codes and sector classification codes.

**Review by maurolepore — CHANGES_REQUESTED** [review]
This `@family` will group this dataset with other [demo datasets](https://2degreesinvesting.github.io/r2dii.dataraw/reference/index.html#section-demo-datasets) but it looks like this dataset doesn't belong there -- as suggested by the lack of the suffix `_demo` in the dataset name.

Instead I think it is more closely related to the [`#' @family datasets for bridging sector classification codes`](https://2degreesinvesting.github.io/r2dii.dataraw/reference/index.html#section-datasets-to-look-up-sector-classification-codes). But that family name is too specific to include `iso_codes` so I changed it to `#' @family datasets to lookup codes` -- which includes datasets too lookup iso-codes and sector-classification-codes.

File: R/iso_codes.R
Comment (maurolepore) [reply R/iso_codes.R:5]: Done this myself, while I was here :). No further action required -- unless my idea is a bad one and you want to revert the relevant commit.

File: data-raw/region_isos.R
Comment (maurolepore) [inline data-raw/region_isos.R:6]: @jdhoffa can you explain why a 2-column dataframe isn't a good alternative? 

A dataframe data structure is most common and easier to work with tidyverse tools. Sure, there is purrr, but not many people is fluent in list-oriented worflows. And it would be easy to extend `data_dictionary()`.

``` r
library(tidyverse)

region_isos_df <- r2dii.dataraw::region_isos %>% 
  purrr::map_df(as_tibble, .id = "region") %>% 
  dplyr::rename(isos = .data$value)

region_isos_df %>% 
  dplyr::group_by(region) %>% 
  # sampling a few rows per group to succinctly show some rows in several groups
  dplyr::filter(dplyr::row_number() %in% 1:3)
#> # A tibble: 72 x 2
#> # Groups:   region [26]
#>    region isos  
#>    <chr>  <chr> 
#>  1 global af    
#>  2 global al    
#>  3 global dz    
#>  4 africa dz    
#>  5 africa ao    
#>  6 africa bj    
#>  7 brazil br    
#>  8 brazil global
#>  9 china  cn    
#> 10 china  hk    
#> # ... with 62 more rows
```

<sup>Created on 2020-02-07 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9001)</sup>

Comment (maurolepore) [reply data-raw/region_isos.R:6]: Maybe what I'm missing is an example showing how you plan to use this dataset.

Comment (maurolepore) [reply data-raw/region_isos.R:6]: And BTW, we are missing an R/regions_isos.R or R CMD check complaints about undocumented data.

File: R/iso_codes.R
Comment (jdhoffa) [reply R/iso_codes.R:5]: Nope, i agree with this and it makes sense to include it with a more broadly defined sector family!

File: data-raw/region_isos.R
Comment (jdhoffa) [reply data-raw/region_isos.R]: Ya so the question here is one of user-friendly vs. computer-friendly. 
To a user, hoping to define a new region, it is most intuitive to say: the region is called "foo" and in it are the countries c("bar1", "bar2", "bar3", ...) (easiest to define as a list i imagine). 
Then, in practice, I agree it would be easeier to have it as a 2-column dataframe. 

Question: If we paste the code you wrote above into the region_isos.R code in data-raw/, can we just have the package do the conversion on build? I will commit what I am talking about, and you can decide if this works or not.

Comment (jdhoffa) [reply data-raw/region_isos.R:6]: I think it's still easiest to just add a new list entry when we want to add a new region (in the data-raw/region_isos.rda list), but I have added a step in the data_prep function in data/raw to convert it to a 2-column dataframe afterward.

Comment (maurolepore) [reply data-raw/region_isos.R:6]: Great. I now understand and I like that you are thinking not only about the user but also about the maintainers.

Comment (jdhoffa) [reply data-raw/region_isos.R:6]: i am both!

**Comment by maurolepore** [comment]
I don't see `region_isos` documented in `data_dictionary()`. Can you add it?


``` r
unique(r2dii.dataraw::data_dictionary()$dataset)
#> [1] "ald"                 "data_dictionary"     "isic_classification"
#> [4] "iso_codes"           "loanbook"            "nace_classification"
#> [7] "overwrite"           "scenario"
```

<sup>Created on 2020-02-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by maurolepore** [comment]
BTW, one data structure that may fit well here is the nested dataframe -- althugh not many people feel comfortable with it.

``` r
library(r2dii.dataraw)
#> Loading required package: r2dii.utils

region_isos %>% 
  tidyr::nest(data = isos)
#> # A tibble: 26 x 2
#>    region                 data              
#>    <chr>                  <list>            
#>  1 global                 <tibble [284 × 1]>
#>  2 africa                 <tibble [58 × 1]> 
#>  3 brazil                 <tibble [2 × 1]>  
#>  4 china                  <tibble [3 × 1]>  
#>  5 eastern_europe_eurasia <tibble [27 × 1]> 
#>  6 eu                     <tibble [29 × 1]> 
#>  7 india                  <tibble [2 × 1]>  
#>  8 japan                  <tibble [2 × 1]>  
#>  9 latin_america          <tibble [46 × 1]> 
#> 10 middle_east            <tibble [13 × 1]> 
#> # … with 16 more rows
```

<sup>Created on 2020-02-10 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by jdhoffa** [comment]
I'm going to add a _super_ unrelated update to `overwrite_demo` while I'm here

**Review by maurolepore — APPROVED** [review]

**Comment by jdhoffa** [comment]
Just a heads up, while I have administrative priveliges on `r2dii.match`, I don't seem to have them on this repo (so I can't merge this PR myself)

**Pull request merged**
